### PR TITLE
Fix user ban modal grammar to account for multiple users

### PR DESCRIPTION
**Pull request by thecashewtrader**
## Description

This PR fixes user ban modal grammar to account for multiple users, addressing #4661 
## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [X] 🙅 no, because this PR does not update server code
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: client/src/app/shared/shared-moderation/user-ban-modal.component.html [hunk 1]
```diff
@@ -27,7 +27,7 @@ <h4 i18n class="modal-title">Ban</h4>
           (click)="hide()" (key.enter)="hide()"
         >
 
-        <input type="submit" i18n-value value="Ban this user" class="peertube-button orange-button" [disabled]="!form.valid" />
+        <input type="submit" i18n-value value="{{ modalMessage }}" class="peertube-button orange-button" [disabled]="!form.valid" />
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-moderation/user-ban-modal.component.html:30]:
```
[value]="submitMessage"
```

File: client/src/app/shared/shared-moderation/user-ban-modal.component.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export class UserBanModalComponent extends FormReactive implements OnInit {
 
   private usersToBan: User | User[]
   private openedModal: NgbModalRef
+  public modalMessage: string = ""
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-moderation/user-ban-modal.component.ts:20]: You don't need `public`, `: string` and prefer using simple quotes `= ''`

File: client/src/app/shared/shared-moderation/user-ban-modal.component.ts [hunk 3]
```diff
@@ -36,6 +37,12 @@ export class UserBanModalComponent extends FormReactive implements OnInit {
   openModal (user: User | User[]) {
     this.usersToBan = user
     this.openedModal = this.modalService.open(this.modal, { centered: true })
+
+    let isSingleUser = true
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-moderation/user-ban-modal.component.ts:41]:
```
const areMultipleUsers = Array.isArray(this.usersToBan) && this.usersToBan.length > 1
```

File: client/src/app/shared/shared-moderation/user-ban-modal.component.ts [hunk 4]
```diff
@@ -36,6 +37,12 @@ export class UserBanModalComponent extends FormReactive implements OnInit {
...
+    if (Array.isArray(this.usersToBan)) {
+      isSingleUser = this.usersToBan.length <= 1
+    }
+    this.modalMessage = isSingleUser ? "Ban this user" : "Ban these users"
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-moderation/user-ban-modal.component.ts:45]:
```
this.submitMessage = areMultipleUsers
  ? $localize`...`
  : $localize`...`
```

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

File: client/src/app/shared/shared-moderation/user-ban-modal.component.ts (hunk 3 above)
Comment (thecashewtrader) [reply client/src/app/shared/shared-moderation/user-ban-modal.component.ts:41]: Won't this cause an error if `usersToBan` is a `User` and not `User[]`?

**Comment by thecashewtrader** [comment]
A pointer: Waiting for a reply/confirmation to https://github.com/Chocobozzz/PeerTube/pull/4722#discussion_r784673561 before changing `isSingleUser` implementation (and/or using `areMultipleUsers` as a variable, though that's trivial).

File: client/src/app/shared/shared-moderation/user-ban-modal.component.ts (hunk 3 above)
Comment (Chocobozzz) [reply client/src/app/shared/shared-moderation/user-ban-modal.component.ts:41]: No because the clause will stop at `Array.isArray` if it's a `User`

**Comment by Chocobozzz** [comment]
Thanks

**Comment by Poslovitch** [comment]
Thanks 👍

**Pull request merged**